src/olympia/versions/models.py

```diff
@@ -1018,7 +1111,14 @@ def should_have_due_date(self):
         """Should this version have a due_date set, meaning it needs a manual review.
 
         See VersionManager.should_have_due_date for logic."""
-        return Version.unfiltered.should_have_due_date().filter(id=self.id).exists()
+        obj = (
+            Version.unfiltered.should_have_due_date(annotate_reasons=True)
+            .filter(id=self.id)
+            .first()
+        )
+        if obj is not None:
+            self._due_date_reasons = obj._due_date_reasons
+        return obj is not None
```

Comment:
tbh I don't like this, because it's setting magical properties on the instance, in a way you may expect from a queryset with annotations, but would by accessing a `@property`.  But we _do_ need the value of `_due_date_reasons` if we're going to update `due_date` immediately after checking if there should be a due_date.  The alternative is we just delete this `@property` and directly call the manager method in the one place it's used (and update a bunch of tests 😒)